### DISCUSSION TITLE
Have we evaluated using Google's Protocol Buffers serialization instead of JSON serialization for cross-bridge communication?

### DISCUSSION THREAD

**Discussion by shudv** [react-native-community/discussions-and-proposals#473] [Ideas]
Have we evaluated using [protocol buffer](https://developers.google.com/protocol-buffers) serialization instead of JSON serialization for cross bridge communicatio in React Native?

I have just begun seriously exploring RN as the choice of framework for one of my critical projects. Based on my very rudimentary understanding, the cross-bridge communication is at the heart of RN. It directly impacts performance and user experience of a RN app. If we use protocol buffer as the serialisation protocol, we get following potential benefits-

1. **40% smaller** serialised messages on an average, which are **HUGE!** savings. This of course does not impact JSI calls as there is no serialisation/deserialization happening there.
2. Out of the box **interoperability** with different kinds of UI layers. This obviously needs more explanation (and I am not even sure if I am thinking in the right direction). But having first class support for protocol buffers means that we can easily author a **single** contract for communciation between native and JS realms and use that contract to generate native classes on all platforms that protocol buffers support.

**Comment by shudv** [comment]
I am slowly catching up on things and it looks like the aim is to get rid of the bridge going forward. In that case, this discussion stands obsolete and we do not need to put much thought into it.

**Reply by cortinico** [reply]
Yup you essentially answered yourself. We're moving towards getting rid of the bridge so we're not looking into changing the serialization mechanism at all.

**Reply by shudv** [reply]
Thanks Nicola! Also, I think the 2nd point of interoperability is directly solved by the CodeGen component in the new architecture that will automatically generate bindings for native-JS crosstalk.
